internal/blocksync/reactor.go

```diff
@@ -186,15 +186,24 @@ func (r *Reactor) OnStop() {
 func (r *Reactor) respondToPeer(ctx context.Context, msg *bcproto.BlockRequest, peerID types.NodeID, blockSyncCh *p2p.Channel) error {
 	block := r.store.LoadBlock(msg.Height)
 	if block != nil {
+		extCommit := r.store.LoadBlockExtCommit(msg.Height)
+		if extCommit == nil {
+			r.logger.Error("peer requesting a block; we have the block but not its extended commit (%v)", block)
+			return fmt.Errorf("blockstore has block but not extended commit %v", block)
+		}
 		blockProto, err := block.ToProto()
 		if err != nil {
-			r.logger.Error("failed to convert msg to protobuf", "err", err)
+			r.logger.Error("failed to convert block to protobuf", "err", err)
```

Comment:
I see that this isn't a change in general, but I think logging and then returnning an error isn't a great pattern (either you handle an error and you should log it there, or you propogate the error up, but if you log and propagate you increase the chance that you'll log it again,

Reply:
See my previous comment. What you say in brackets is generally true, but might break down in the presence of a bug... and you might find yourself with _no_ logs, rather than two, when chasing _that_ bug

Reply:
In this case we are logging the error already further up the call hierarchy (this method's only called in one function, which is called in turn within another function, which in turn logs the error). We can rather wrap the error in the descriptive message and pass it up the chain.